### PR TITLE
164 Add the list of all possible methods (blocks) for ReportCard 

### PR DESCRIPTION
**Pull request by kartikeyakirar**
fixes https://github.com/insightsengineering/teal.reporter/issues/164

for reviewer
- [ ] Confirm that the title/filename accurately reflects the content and scope.
- [ ] Verify that all available content blocks in teal.reporter are listed in the table.
- [ ] Any suggestion on changes in table . (colnames title if needs changes.)
- [ ] Read through the vignette for flow and coherence; 
- [ ] Diagram shows accurate description.

### CODE REVIEW

**Comment by kartikeyakirar** [comment]
I have read the CLA Document and I hereby sign the CLA

File: vignettes/teal-reporter-blocks-overview.Rmd [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+## Table: Content Blocks in `teal.reporter`
+The following table outlines the different blocks that can be included in a `ReportCard`, 
+along with descriptions and usage examples:
+
+| **Block Type** | **Description** | **Usage Example** |
+  |----------------|----------------|-------------------|
+  | **`ReportCard`** | Combines various content blocks into a single card. | `report_card <- ReportCard$new()` |
+  | **`TextBlock`** | Adds text-based content to the report. | `report_card$append_text(<text>)` |
+  | **`PictureBlock`** | Contains graphical content like plots or diagrams. | `report_card$append_plot(<plot>)` |
```
[the remark this answers is not in the document]
Comment (kartikeyakirar) [reply vignettes/teal-reporter-blocks-overview.Rmd:22]: In `PictureBlock` when we talk about graphical content, its supported type is specifically designed to handle plot objects like "ggplot", "grob", "trellis", and "Heatmap". 
https://github.com/insightsengineering/teal.reporter/blob/main/R/PictureBlock.R#L46-L66

Class accepts a supported plot object and creates a temporary file to include it in the report. 
https://github.com/insightsengineering/teal.reporter/blob/main/R/PictureBlock.R#L122

So I believe it should be graphical content but I will mention specific accepted type.

File: vignettes/images/blocks.png
Comment (pawelru) [inline vignettes/images/blocks.png:1]: would it be possible to provide a code for future edits?
image files are not super convenient with regard to potential updates

Comment (kartikeyakirar) [reply vignettes/images/blocks.png:1]: I have added mermaid diagram 
image

Comment (kartikeyakirar) [reply vignettes/images/blocks.png:1]: [c961007](https://github.com/insightsengineering/teal.reporter/pull/272/commits/c96100772b36be2ebb938f4dd1fd7e2e898221b1)

File: vignettes/teal-reporter-blocks-overview.Rmd [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+## Overview of Content Blocks
+This document serves as a comprehensive guide to the various types of content blocks available in the `teal.reporter`. 
+These blocks allow users to structure and customize reports.
+
+## Table: Content Blocks in `teal.reporter`
+The following table outlines the different blocks that can be included in a `ReportCard`, 
+along with descriptions and usage examples:
+
+| **Block Type** | **Description** | **Usage Example** |
```
[the remark this answers is not in the document]
Comment (kartikeyakirar) [reply vignettes/teal-reporter-blocks-overview.Rmd:18]: [60fd1b1](https://github.com/insightsengineering/teal.reporter/pull/272/commits/60fd1b1033fc247fef5d4d6809cb7693aaf13000)

File: vignettes/images/blocks.png
Comment (pawelru) [reply vignettes/images/blocks.png:1]: Thanks!

File: vignettes/teal-reporter-blocks-overview.Rmd [hunk 3]
```diff
@@ -0,0 +1,120 @@
+---
+title: "Comprehensive Overview of Content Blocks in `teal.reporter`"
```
Comment (gogonzo) [inline vignettes/teal-reporter-blocks-overview.Rmd:2]: When making pages filenames should be the same as titles. If they are different then it is more difficult to navigate in the project when developing.

Comment (kartikeyakirar) [reply vignettes/teal-reporter-blocks-overview.Rmd]: done [d70eefe](https://github.com/insightsengineering/teal.reporter/pull/272/commits/d70eefe8976abb78d0ef76a14ba3637e8dd5efcf)

**Comment by kartikeyakirar** [comment]
> It would be great to link it somewhere in the roxygen docs of the described classes. I'm just concerned that this would not be ever looked into without such refs.

I agree and have linked this to ReportCard since all the blocks are utilized within it. This seems like the most appropriate place, especially considering it's an exported class. here [26b13de](https://github.com/insightsengineering/teal.reporter/pull/272/commits/26b13decbaf14d78939355353a8ae49e816866ca)

**Review by gogonzo — APPROVED** [review]
👍

**Pull request merged**